lib/sip2/messages/base.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Sip2
+  module Messages
+    #
+    # Sip2 Base message handler
+    #
+    class Base
+      @message_class_lookup = {}
+
+      def self.message_class_for_method(method_name) # rubocop:disable Metrics/MethodLength
+        return @message_class_lookup[method_name] if @message_class_lookup.key? :method_name
+
+        @message_class_lookup[method_name] =
+          begin
+            class_name = method_name.to_s.capitalize.gsub(%r{(?:_|(/))([a-z\d]*)}i) do
```

Comment:
Might be good to include a comment explaining what you are doing with these regexes or move the code into a named method to make it clearer.

Reply:
this was based heavily on the Rails `classify` method. But sure, comment would help